modules/core/src/lib/attribute-transition-manager.js

```diff
@@ -213,16 +211,23 @@ export default class AttributeTransitionManager {
       });
     }
 
-    transition.attributeInTransition.update({buffer});
+    transition.length = toLength;
```

Comment:
Do we have to keep track of the length like this? Isn't this the same as `transition.buffer`'s length?

Reply:
We only ever expand the buffers when the sizes are too small. So they can be over-allocated.

Reply:
Why do we only update the buffer size when it's too small? Wouldn't it be simpler (and match reader/developer expectations) to correctly size the buffer whether it has grown or shrunk in length?

Reply:
Allocating memory for a buffer has a performance cost. We do the same thing in the Attributes class (over-allocate then use `bufferSubData`). Generally speaking we should never expect the buffers in deck.gl sized precisely to the draw list.